### PR TITLE
CM-872 IgnorePropertyInSwagger

### PR DESCRIPTION
**Pull request by GergoVandor**

### CODE REVIEW

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace IgnorePropertyInSwagger.SchemaFilters;
+
+public class SwaggerIgnoreFilter : ISchemaFilter
+{
+    public void Apply(OpenApiSchema schema, SchemaFilterContext schemaFilterContext)
+    {
+        if (schema.Properties.Count == 0)
+            return;
```
Comment (mahrncic) [inline aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs:15]: Always use curly braces even though there is only one line inside the `if` statement

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        var properties = schemaFilterContext.Type.GetProperties();
+
+        var excludedList = properties
+            .Where(m => m.GetCustomAttribute<SwaggerIgnoreAttribute>() is not null)
+            .Select(m => m.GetCustomAttribute<JsonPropertyNameAttribute>()?.Name ?? m.Name.ToCamelCase());
+
+        foreach (var excludedName in excludedList)
+        {
+            if (schema.Properties.ContainsKey(excludedName))
+                schema.Properties.Remove(excludedName);
```
Comment (mahrncic) [inline aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs:26]: Same here, please add curly braces.

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+public static class StringExtensions
+{
+    public static string ToCamelCase(this string input)
+    {
+        if (string.IsNullOrEmpty(input) || char.IsLower(input[0]))
+            return input;
+
+        var chars = input.ToCharArray();
+        chars[0] = char.ToLower(input[0]);
+        return new(chars);
```
Comment (mahrncic) [inline aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs:12]: Always separate a `return` statement with a blank line between the `return` and the rest of the code

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+namespace IgnorePropertyInSwagger.Extensions;
+
+public static class StringExtensions
+{
+    public static string ToCamelCase(this string input)
+    {
+        if (string.IsNullOrEmpty(input) || char.IsLower(input[0]))
+            return input;
```
Comment (mahrncic) [inline aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs:8]: Curly braces

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Endpoints/WeatherForecasts.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            {
+                var forecast = Enumerable.Range(1, 5).Select(index =>
+                        new WeatherForecast
+                        (
+                            DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
+                            Random.Shared.Next(-20, 55),
+                            Summaries[Random.Shared.Next(Summaries.Length)]
+                        ))
+                    .ToArray();
+                return forecast;
```
Comment (mahrncic) [inline aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Endpoints/WeatherForecasts.cs:24]: Add a blank line before the `return` statement

**Review by mahrncic — CHANGES_REQUESTED** [review]

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs (hunk 1 above)
Comment (GergoVandor) [reply aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs]: done

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs (hunk 2 above)
Comment (GergoVandor) [reply aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/SchemaFilters/SwaggerIgnoreFilter.cs]: done

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs (hunk 3 above)
Comment (GergoVandor) [reply aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs:15]: done

File: aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs (hunk 4 above)
Comment (GergoVandor) [reply aspnetcore-webapi/IgnorePropertyInSwagger/IgnorePropertyInSwagger/Extensions/StringExtensions.cs:9]: done

**Review by mahrncic — APPROVED** [review]

**Review by mahrncic — APPROVED** [review]

**Pull request merged**